.azure/pipelines/Phonebook.preview.yml

```diff
@@ -0,0 +1,193 @@
+pr: 
+  branches:
+    include:
+      - master
+
+variables:
+  image_repo: danielhabenicht/phonebook
+  image_tag: pr-$(System.PullRequest.PullRequestNumber) 
+  helm_artifact_name: HelmPhonebookPackage
+
+  jobs:
+    - job: build_phonebook_frontend
+      displayName: 'Phonebook Frontend'
+      pool:
+        vmImage: 'ubuntu-16.04'
+      steps:
+      - task: UseNode@1
+        inputs:
+          version: '11.x'
+
+      - script: node version.js $(image_tag) $(Build.SourceVersion) $(Build.SourceVersion)
+        displayName: 'Write Version Number'    
+        workingDirectory: 'Phonebook.Frontend/'
+
+      - script: docker build -t $(image_repo):$(image_tag) .
+        displayName: 'Build Image'              
+        workingDirectory: 'Phonebook.Frontend/'
+      
+      - script: |
+          docker login -u $DOCKER_REGISTRY_USER -p $(DOCKER_REGISTRY_PASSWORD)
+          docker push $(image_repo):$(image_tag)
+        workingDirectory: 'Phonebook.Frontend/'
+        displayName: 'Push Image'                      
+        env:
+          DOCKER_REGISTRY_PASSWORD: $(DOCKER_REGISTRY_PASSWORD)
+          DOCKER_REGISTRY_USER: $(DOCKER_REGISTRY_USER)
+
+    - job: build_phonebook
+      displayName: Phonebook
+      pool:
+        vmImage: 'ubuntu-16.04'
+      steps:
+      - task: HelmInstaller@1
+        inputs:
+          helmVersionToInstall: 'latest'
+      - script: helm dep update
+        workingDirectory: 'Phonebook/phonebook/'
+      - task: HelmDeploy@0
+        displayName: Helm package
+        inputs:
+          command: package
+          chartPath: Phonebook/phonebook
+          destination: $(Build.ArtifactStagingDirectory)
+      - publish: $(Build.ArtifactStagingDirectory)
+        artifact: $(helm_artifact_name)
+        
+    - job: deploy_preview
```

Comment:
it's wired that the build deploys something. Maybe @kirkone has an idea.

Reply:
seems like he is out of ideas 😄